core/computed/entity-classification.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * @license Copyright 2022 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import {makeComputedArtifact} from './computed-artifact.js';
+import {NetworkRecords} from './network-records.js';
+import {Util} from '../util.cjs';
+import UrlUtils from '../lib/url-utils.js';
+import thirdPartyWeb from '../lib/third-party-web.js';
+
+/** @typedef {Map<string, LH.Artifacts.Entity>} EntityCache */
+
+class EntityClassification {
+  /**
+   * @param {EntityCache} entityCache
+   * @param {string} url
+   * @return Entity | undefined
+   */
+  static makeUpAnEntity(entityCache, url) {
+    if (!UrlUtils.isValid(url)) return;
+    // We can make up an entity only for those URLs with a valid domain attached.
+    // So we further restrict from allowed URLs to (/^h/ or http[s] and h2).
+    if (!(UrlUtils.isValid(url) && UrlUtils.isProtocolAllowed(url) &&
+      Util.createOrReturnURL(url).protocol[0] === 'h')) return;
```

Comment:
excellent until the day we decide to support [`hcp`](https://en.wikipedia.org/wiki/List_of_URI_schemes#:~:text=host%3E%5B%3A%3Cport%3E%5D%5B%3B%3Cparameters%3E%5D-,hcp,-Displaying%20a%20help) :P 

Just kidding, would be a bit more comfortable just doing `startsWith('http')`.  maybe whole thing could be:

```
!['http', 'https'].includes(Util.createOrReturnURL(url).protocol)) return;
```

Because the former line is already checking the url is valid.

Reply:
lol, good catch! I think I confused this one with CDP's protocol field, I guess. Otherwise no reason to expect `h2` there.

Reply:
Updated with 

```
if (!Util.createOrReturnURL(url).protocol.startsWith('http')) return;
```